src/app-layer.c

```diff
@@ -80,6 +91,14 @@ static inline int ProtoDetectDone(const Flow *f, const TcpSession *ssn, uint8_t
             (FLOW_IS_PM_DONE(f, direction) && FLOW_IS_PP_DONE(f, direction)));
 }
 
+static void AppLayerIncFlowCounter(ThreadVars *tv, uint8_t ipproto, AppProto alproto)
```

Comment:
If you just pass 'Flow *f' to this function you can directly use f->protomap and f->alproto. Safes one FlowGetProtoMapping call.